crepecake/src/test/java/com/google/cloud/tools/crepecake/image/ImageLayersTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.image;
+
+import com.google.cloud.tools.crepecake.blob.BlobDescriptor;
+import java.util.Arrays;
+import java.util.List;
+import org.hamcrest.CoreMatchers;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+/** Tests for {@link ImageLayers}. */
+public class ImageLayersTest {
+
+  @Mock private DescriptorDigest mockDescriptorDigest1;
```

Comment:
This is a lot of mocks. Could any of these be replaced by actual objects with known values?

Reply:
Is having a lot of mocks bad practice? I needed many different blob descriptors because ImageLayers prevents the duplicate layer digests.

Reply:
Not in and of itself, but the more you mock, the less you test. In general test data that can be an actual object, should be an actual object. I'm not sure if that applies here or not.

Also, not all mocks have to be fields. Mocks referenced in only one method should be local variables. Here it looks like the descriptor digests can all move into `setUpFakes`.